src/app/base/hooks/base.ts

```diff
@@ -175,3 +175,24 @@ export const useScrollToTop = (): void => {
     window.scrollTo(0, 0);
   }, [pathname]);
 };
+
+export const usePreviousPersistent = <T extends unknown>(
```

Comment:
This could do with a docstring and some comments below.

Reply:
I think it's worth adding these docstrings as it's surprising how easy it is to come back and not remember what's doing what.

I think some comments inside the hook would be worth adding as well to help follow the logic.